tools/docgenerator.js

```diff
@@ -560,17 +560,18 @@ function collectDocs (functionNames, inputPath) {
     if (path.indexOf('docs') === -1 && functionIndex !== -1) {
       if (path.indexOf('expression') !== -1) {
         category = 'expression'
-      } else if (/^.\/lib\/type\/[a-zA-Z0-9_]*\/function/.test(fullPath)) {
+      } else if (/\/lib\/cjs\/type\/[a-zA-Z0-9_]*\/function/.test(fullPath)) {
+        // for type/bignumber/function/bignumber.js, type/fraction/function/fraction.js, etc
         category = 'construction'
-      } else if (/^.\/lib\/core\/function/.test(fullPath)) {
+      } else if (/\/lib\/cjs\/core\/function/.test(fullPath)) {
         category = 'core'
       } else {
         category = path[functionIndex + 1]
       }
-    } else if (fullPath === './lib/cjs/expression/parse.js') {
+    } else if (fullPath.endsWith('/lib/cjs/expression/parse.js')) {
       // TODO: this is an ugly special case
       category = 'expression'
```

Comment:
Why not just move `expression/parse.js` to `expression/function/parse.js`? It seems like it makes more sense in that location anyway, it only requires changing a very small handful of source files, and would remove this "ugly special case" altogether.